File: components/bottom-sheet/flutter/ciata_bottom_sheet.dart
import 'package:flutter/material.dart';

/// Implementação experimental do CMP-0018 Bottom Sheet para Flutter.
Future<T?> showCiataBottomSheet<T>({
  required BuildContext context,
  required String title,
  required Widget child,
  String closeLabel = 'Fechar',
}) {
  final trimmedTitle = title.trim();
  final trimmedCloseLabel = closeLabel.trim();
  if (trimmedTitle.isEmpty) {
    throw ArgumentError.value(title, 'title', 'não pode ser vazio');
  }
  if (trimmedCloseLabel.isEmpty) {
    throw ArgumentError.value(closeLabel, 'closeLabel', 'não pode ser vazio');
  }

  return showModalBottomSheet<T>(
    context: context,
    isScrollControlled: true,
    builder: (context) => SafeArea(
      child: Padding(
        padding: const EdgeInsets.all(16),
        child: Column(
          mainAxisSize: MainAxisSize.min,
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Semantics(header: true, child: Text(trimmedTitle)),
            const SizedBox(height: 16),
            Flexible(child: SingleChildScrollView(child: child)),
            const SizedBox(height: 16),
            TextButton(
              onPressed: () => Navigator.of(context).pop(),
              child: Text(trimmedCloseLabel),
            ),
          ],
        ),
      ),
    ),
  );
}

File: components/bottom-sheet/android/CiataBottomSheet.kt
package org.ciata.ds.bottomsheet

import androidx.compose.material3.ModalBottomSheet
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.semantics.heading
import androidx.compose.ui.semantics.semantics

/** Implementação experimental do CMP-0018 Bottom Sheet para Jetpack Compose. */
@Composable
fun CiataBottomSheet(
    open: Boolean,
    title: String,
    onDismissRequest: () -> Unit,
    closeLabel: String = "Fechar",
    content: @Composable () -> Unit,
) {
    val normalizedTitle = title.trim()
    val normalizedCloseLabel = closeLabel.trim()

    require(normalizedTitle.isNotEmpty()) { "title não pode ser vazio." }
    require(normalizedCloseLabel.isNotEmpty()) { "closeLabel não pode ser vazio." }
    if (!open) return

    ModalBottomSheet(onDismissRequest = onDismissRequest) {
        Text(normalizedTitle, modifier = Modifier.semantics { heading() })
        content()
        TextButton(onClick = onDismissRequest) {
            Text(normalizedCloseLabel)
        }
    }
}
